### PR TITLE
Reference the class-based schema

### PR DESCRIPTION
**Pull request by davidw**
https://github.com/exAspArk/batch-loader/issues/48

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -291,6 +291,13 @@ Schema = GraphQL::Schema.define do
   use BatchLoader::GraphQL
 end
 ```
+or, using the class-based (introduced in graphql-ruby in version 1.8) way of defining a schema:
+```
```
Comment (exAspArk) [inline README.md]: Could you please append `rb` to enable syntax highlighting?

**Comment by davidw** [comment]
I could do that if you'd prefer a rewrite to duplicating everything with the class-based version. I think that'd be tidier.

**Comment by exAspArk** [comment]
Yeah, that would be amazing if you could rewrite the examples to use class-based schema definitions! 🙏

**Comment by bbugh** [comment]
Thank you for doing this! As you mentioned in #48, I had some concern that this library didn't support the class-based API.

**Comment by bbugh** [comment]
Unfortunately, I noticed that it's incorrect for graphql-ruby current API. All of the field lines are using the deprecated API to describe types and fields, [which would cause an error without activating the `GraphQL::DeprecatedDSL` flag](https://graphql-ruby.org/schema/class_based_api.html#list-types-and-non-null-types).

Current PR:

```ruby
class QueryType < Types::BaseObject
  #                   this is the old API style typing, which breaks without
  #             vvvvv "using GraphQL::DeprecatedDSL" refinement
  field :posts, !types[PostType], resolve: ->(obj, args, ctx) { Post.all }
  #                               ^^^ resolve does still work, but it's not "official"
end
```

[officially recommended way to write this](https://graphql-ruby.org/fields/introduction.html):

```ruby
class QueryType < Types::BaseObject
  field :posts, [PostType], null: false

  def posts
    Post.all
  end
end
```

I'm happy to offer specific corrections as review comments in the PR if desired.

**Comment by bbugh** [comment]
My offer to get this corrected and launched is still available!

**Comment by davidw** [comment]
@bbugh any contributions are more than welcome!

File: README.md [hunk 2]
```diff
@@ -235,23 +235,37 @@ Batching is particularly useful with GraphQL. Using such techniques as preloadin
 Let's take a look at the simple [graphql-ruby](https://github.com/rmosolgo/graphql-ruby) schema example:
 
 ```ruby
-Schema = GraphQL::Schema.define do
-  query QueryType
+class MyprojectSchema < GraphQL::Schema
```
Comment (exAspArk) [inline README.md]: could we make the naming a bit more consistent`MyProjectSchema` or simply `Schema`

File: README.md [hunk 3]
```diff
@@ -235,23 +235,37 @@ Batching is particularly useful with GraphQL. Using such techniques as preloadin
...
+
+  use BatchLoader::GraphQL
 end
 
-QueryType = GraphQL::ObjectType.define do
-  name "Query"
-  field :posts, !types[PostType], resolve: ->(obj, args, ctx) { Post.all }
+module Types
+  class QueryType < Types::BaseObject
+    field :posts, ![PostType]
```
Comment (exAspArk) [inline README.md]: could we use [null: false](https://graphql-ruby.org/type_definitions/objects) vs `!`?


here is an example from the tests:

https://github.com/exAspArk/batch-loader/blob/9b015fb3f57c8783d7c3e780ed5596b46e1dbea2/spec/fixtures/graphql_schema.rb#L33-L66

File: README.md [hunk 4]
```diff
@@ -235,23 +235,37 @@ Batching is particularly useful with GraphQL. Using such techniques as preloadin
 Let's take a look at the simple [graphql-ruby](https://github.com/rmosolgo/graphql-ruby) schema example:
 
 ```ruby
-Schema = GraphQL::Schema.define do
-  query QueryType
+class MyprojectSchema < GraphQL::Schema
+  mutation(Types::MutationType)
+  query(Types::QueryType)
+
+  use BatchLoader::GraphQL
```
Comment (exAspArk) [inline README.md]: could we remove the line and include it in the second example without N+1 queries?

File: README.md [hunk 5]
```diff
@@ -286,7 +304,7 @@ end
 And setup GraphQL to use the built-in `lazy_resolve` method:
 
 ```ruby
-Schema = GraphQL::Schema.define do
+class GraphqlSchema < GraphQL::Schema
```
Comment (exAspArk) [inline README.md]: `GraphqlSchema` could we call it the same way as in the first example? so people understand what has to be changed in the first example to get to the second one

File: README.md (hunk 2 above)
Comment (davidw) [reply README.md]: I like to use names that are really obviously not a 'magic' name, but something that can be changed as desired.

**Comment by bbugh** [comment]
It all looks correct now, nice work! :tada:

**Comment by exAspArk** [comment]
@davidw excellent, thanks a lot! 🙌

**Pull request merged**